plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/delete/LazyTrinoRow.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.delete;
+
+import io.trino.spi.Page;
+import io.trino.spi.type.Type;
+import org.apache.iceberg.StructLike;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.trino.plugin.iceberg.IcebergPageSink.getIcebergValue;
+import static java.util.Objects.requireNonNull;
+
+final class LazyTrinoRow
```

Comment:
add `@see TrinoRow` 
and `@NotThreadSafe`

Reply:
Technically, this class is thread safe. It's similar to the `hash` field of `java.lang.String`. Writes to references are atomic, and the computed values are always same, so the race is benign and simply results in duplicate work.

I'm not going to add these annotations since we're only using these in a single context of `ConnectorPageSource` and thus discussion about thread safety seems confusing. Readers might wonder why we are talking about that and if they're missing some other usage.